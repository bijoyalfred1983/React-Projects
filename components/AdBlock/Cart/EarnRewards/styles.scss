@import '../../../theme/variables';


.earnRewardsContainer {
  width: 100%;
  margin: 68px 0;

  .earnRewardsBox {
    width: 75%;
    display: flex;
    margin: auto;
  }

  .imageContent {
    margin-right: 32px;
  }

  .AZ_Rewards_Card {
    width: 146px;
    border-radius: 0.5px;
    vertical-align: middle;
  }

  .rewardsText {
    width: 417px;
    font-family: $font-az-medium;
    font-size: 24px;
    font-weight: 500;
    line-height: 1;
  }

  .rewards_balance_heading {
    padding-left: 10px;
  }

  .rewardsFormatText {
    text-transform: uppercase;
  }
  
  .qualifilesText {
    font-family: $font-az-app;
    font-size: 16px;
    color: $grey-ultra-dark;
    line-height: 20px;
    letter-spacing: 0.2px;
    margin-top: 7px; 

    a {
      color: $grey-ultra-dark;
      font-family: $font-category-cont;
      text-decoration: underline;      
      font-weight: 550;
      font-size: 15px;

      @media #{$extra-large-and-down} {
        font-size: 14px;
      }
    }
  } 

  .fullStopSymbol {
    padding-left: 1px;
    font-weight: 550;
  }

  .contentFlex {
    flex: 1;
	
    @media #{$medium-only} {
      padding-left: 60px;
    }	
  }
}

.superScript{
  top:-9px;
  font-size:45%;
}

@media #{$extra-large-and-up} {
  .earnRewardsContainer {
    .rewardsTextMob {
      display: none;
    }
  }
}

@media #{$extra-large-and-down} {
  .earnRewardsContainer {
    margin: 0;
    border-top: 1px solid $color-lighter-grey;  
    width: auto;

    .rewardsText {
      display: none;
    }

    .imageContent {
      margin-right: 16px;
    }

    .AZ_Rewards_Card {
      width: 45px;
      border-radius: 0.3px;
    }

    .rewardsTextBoxMob {
      margin: auto;
    }

    .earnRewardsBox {
      width: 100%;
      display: flex;
      margin: auto;
      flex-flow: column;
      background-color: #fff;
      padding: 39px 16px;
	  
      @media #{$medium-only} {
        padding: 20px 0 10px;
      }	  
    }

    .rewardsTextMob {
      width: 229px;
      font-family: $font-az-medium;
      font-size: 20px;
      font-weight: 550;
      line-height: 1.2;
	  
      @media #{$medium-only} {
        font-size: 15px;	  
      }	  
    }

    .qualifilesText {
      font-size: 14px;
      line-height: 1.29;
      text-align: center;
	  
      @media #{$medium-only} {
        text-align: justify;		
        font-size: 12px;
        color: $grey-ultra-dark;
        margin-top: -5px;
      }	  
    }

    .loginText {
      font-size: 14px;
      line-height: 1.29;
    }
  }
}

@media #{$medium-only} {
  .earnRewardsContainer {
    margin: 0;
    border-top: 1px solid $color-lighter-grey;  
    margin-top: 35px;
    width: auto;
	
    .earnRewardsBox {
      padding-top: 10px;
	  
      .rewardsTextBoxMob {	
        .imageContent {
          margin-top: 10px;
        }	  
      }
    }
  }
}

@media print{
  .earnRewardsContainer{
    margin: 75px 0;
    clear: both;
  }
}
