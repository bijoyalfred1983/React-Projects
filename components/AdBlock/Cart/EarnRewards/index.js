/* @flow */

import React from 'react';
import styles from './styles.scss';
import { fetchStaticLabel } from '../../../containers/App';

type Props = {
  data: Object,
  tablet: boolean,
  staticLabelData: Object
};

const anonymousUserMsg = (rewardsInfo, staticLabel) => {
  if (rewardsInfo.rewardsCreditEligible) {
    return (
      <div className={styles.qualifilesText}>
        <span>
          {fetchStaticLabel(
            'label_cart_rewards_ThisOrderQualifiesForRewardsToClaimTheEarnCredits',
            staticLabel
          )}
        </span>
        <a href="/signin">
          {fetchStaticLabel('label_cart_rewards_LogIn', staticLabel)}
        </a>
        <span>
          &nbsp;{fetchStaticLabel(
            'label_cart_rewards_toYourExistingAccountOrSignnupAfterCheckout',
            staticLabel
          )}
        </span>
      </div>
    );
  }
  return (
    <div className={styles.qualifilesText}>
      <span>
        This order does not qualify for a rewards credit. Spend an additional $
      </span>
      {rewardsInfo.requiredAmountToQualify}
      <span> to qualify. To claim and earn credits </span>
      <a href="/signin">
        {fetchStaticLabel('label_cart_rewards_LogIn', staticLabel)}
      </a>
      <span>
        &nbsp;{fetchStaticLabel(
          'label_cart_rewards_toYourExistingAccountOrSignnupAfterCheckout',
          staticLabel
        )}
      </span>
    </div>
  );
};

const loggedInUserMsg = (rewardsInfo, staticLabel) => {
  if (rewardsInfo.rewardsAccountLinked) {
    if (rewardsInfo.rewardsCreditEligible) {
      if (parseFloat(rewardsInfo.rewardsBalanceInfo.rewardBalance) > 0) {
        return (
          <div className={styles.qualifilesText}>
            {fetchStaticLabel('label_checkout_os_YouHave', staticLabel)} ${
              rewardsInfo.rewardsBalanceInfo.rewardBalance
            }{' '}
            rewards rewards credit on your account #{
              rewardsInfo.rewardsBalanceInfo.loyaltyCardNumber
            }. During checkout you can use your rewards credit to pay for your
            order.
          </div>
        );
      }
      return (
        <div className={styles.qualifilesText}>
          You have ${rewardsInfo.rewardsBalanceInfo.rewardBalance} rewards
          credit on your account #{
            rewardsInfo.rewardsBalanceInfo.loyaltyCardNumber
          }. This order qualfies for rewards points that can be used towards
          your next order.
        </div>
      );
    }
    return (
      <div className={styles.qualifilesText}>
        You have ${rewardsInfo.rewardsBalanceInfo.rewardBalance} rewards credit
        on your account #{rewardsInfo.rewardsBalanceInfo.loyaltyCardNumber}.
        This order does not qualify for a rewards credit. Spend an additional ${
          rewardsInfo.requiredAmountToQualify
        }{' '}
        to qualify.
      </div>
    );
  }
  return (
    <div className={styles.qualifilesText}>
      Doesnâ€™t look like you have rewards linked with your My Zone account. Enter
      an existing rewards number at checkout or register for rewards at order
      confirmation to claim rewards credit for this order.
    </div>
  );
};

const renderMessage = (rewardsInfo, staticLabel) => {
  if (rewardsInfo.loggedInUser) {
    return loggedInUserMsg(rewardsInfo, staticLabel);
  }
  return anonymousUserMsg(rewardsInfo, staticLabel);
};

export default (props: Props) => {
  const rewardsInfo = props.data;
  const balance =
    rewardsInfo.rewardsBalanceInfo &&
    rewardsInfo.rewardsBalanceInfo.rewardBalance
      .toFixed(2)
      .toString()
      .split('.');
  const staticLabel = props.staticLabelData ? props.staticLabelData : {};
  const lblEarnAutozoneRewards = fetchStaticLabel(
    'label_cart_rewards_EarnAutozoneRewards',
    staticLabel
  );
  return (
    <div
      style={{
        display: props.tablet && props.tablet === true ? 'none' : 'block'
      }}
      className={styles.earnRewardsContainer}
    >
      <div className={styles.earnRewardsBox}>
        <div className={styles.rewardsTextBoxMob}>
          <span className={styles.imageContent}>
            <img
              className={styles.AZ_Rewards_Card}
              src="/images/autoZoneRewards.png"
              alt="AZ_rewards"
            />
          </span>
          <span
            className={`${styles.rewardsTextMob} ${styles.rewardsFormatText}`}
          >
            <span>{lblEarnAutozoneRewards}</span>
          </span>
        </div>
        <div className={styles.contentFlex}>
          <div className={styles.rewardsText}>
            <span>{lblEarnAutozoneRewards}</span>
            {rewardsInfo.loggedInUser &&
              balance &&
              balance.length > 0 && (
                <span className={styles.rewards_balance_heading}>
                  <sup className={styles.superScript}>$</sup>
                  {balance[0]}
                  <sup className={styles.superScript}>{balance[1]}</sup>
                </span>
              )}
          </div>
          {renderMessage(rewardsInfo, staticLabel)}
        </div>
      </div>
    </div>
  );
};
