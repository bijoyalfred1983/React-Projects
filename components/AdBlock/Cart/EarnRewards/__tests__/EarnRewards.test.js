import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import EarnRewards from '../';

describe('<EarnRewards />', () => {
  test('renders correctly EarnRewards', () => {
    const wrapper = mount(<EarnRewards />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
