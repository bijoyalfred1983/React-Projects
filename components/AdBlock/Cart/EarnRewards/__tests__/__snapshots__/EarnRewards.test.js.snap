// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EarnRewards /> renders correctly EarnRewards 1`] = `
<Component>
  <div
    className="earnRewardsContainer"
  >
    <div
      className="earnRewardsBox"
    >
      <div
        className="rewardsTextBoxMob"
      >
        <div
          className="imageContent"
        >
          <img
            alt="AZ_rewards"
            className="AZ_Rewards_Card"
            src="/images/autoZoneRewards.png"
          />
        </div>
        <div
          className="rewardsTextMob rewardsFormatText"
        >
          Earn AutoZone® Rewards
        </div>
      </div>
      <div
        className="contentFlex"
      >
        <div
          className="rewardsText rewardsFormatText"
        >
          Earn AutoZone® Rewards
        </div>
        <div
          className="qualifilesText"
        >
          This order qualifies for Rewards! To clain the earn credits, 
          <a
            className="loginText"
            href="http://localhost/auth/oauth/v2/authorize?client_id=l7xx85b096d1f8c3426cb3ed7724c7863444&response_type=code&redirect_uri=https://example.com&scope=ecommb2c"
          >
            LOG IN
          </a>
           to your existing account or sign-up after checkout.
        </div>
      </div>
    </div>
  </div>
</Component>
`;
